### PR TITLE
Fix(DG): Update user story formatting

### LINKED ISSUE #354: DG: Ensure all user stories have all 3 parts

### PR DESCRIPTION
**Pull request by DrWala**
## What this does
- Closes #354
- Removes user stories we don't need
- Updates the word "task" to "issue"
- Ensures all user stories have all 3 parts

### What needs to be discussed
- Phrasing for the 3 "find" stories. Are they ok? Are they too complex?

## How to test
1. cd to `docs/` folder of the project.
2. Run `bundle exec jekyll serve`.
3. Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
4. Check that the DG has the changes listed above under the "User Stories" section

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -602,29 +602,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | - | - | - | - |
-| `* * *` | new user | see usage instructions | refer to instructions if I forget how to use the app |
 | `* * *` | confused user | have a help command | learn how to use the application |
-| `* * *` | user | add a new person | |
-| `* * *` | user | add issues | |
-| `* * *` | user | see a list of pending tasks sorted by deadline | prioritise which to do |
-| `* * *` | user | search for tasks | |
+| `* * *` | user | add issues | track associated issues with rooms and their status|
+| `* * *` | user | see a list of pending issues sorted by deadline | prioritise which issue to work on |
+| `* * *` | user | search for issues | find a specific task and get its information or tasks matching a specific criteria|
```
Comment (colintkn) [inline docs/DeveloperGuide.md:608]: Maybe want to break it up ? 
| `* * *` | user | search for issues | find a specific task and get its information |
| `* * *` | user | search for issues | find tasks matching a specific criteria |

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -602,29 +602,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * *` | user | close issues | focus on open issues |
-| `* * *` | user | update issues | |
-| `* * *` | user | remove issues | |
-| `* * *` | user | add residents | |
-| `* * *` | user | search for residents | |
+| `* * *` | user | update issues | ensure it has the latest and correct information |
+| `* * *` | user | remove issues | not worry about issues that are no longer relevant |
+| `* * *` | user | add residents | track and manage residents who are staying or want to stay in the hostel |
+| `* * *` | user | search for residents | find a specific resident's information or residents matching a specific criteria |
```
Comment (colintkn) [inline docs/DeveloperGuide.md:613]: Change to: 
| * * * | user | search for residents | find a a specific resident's information |
Can remove  " .... residents matching a specific criteria". 

Rationale:
rfind only supports searching based on name. (can't find email etc)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DrWala) [reply docs/DeveloperGuide.md:608]: Good idea, thanks

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (DrWala) [reply docs/DeveloperGuide.md:613]: Okie done, thanks

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (stein414) [inline docs/DeveloperGuide.md:608]: Add the spaces before the last |

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -602,29 +602,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * *` | user | update resident details | ensure their information is up to date |
-| `* * *` | user | remove residents | |
-| `* * *` | user | add new rooms | |
+| `* * *` | user | remove residents | stop tracking residents who are not in the hostel or don't wish to stay anymore |
+| `* * *` | user | add new rooms | track and manage rooms in the hostel|
```
Comment (stein414) [inline docs/DeveloperGuide.md:616]: Add the spaces before the last |

Comment (DrWala) [reply docs/DeveloperGuide.md]: Done

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DrWala) [reply docs/DeveloperGuide.md]: Done

**Review by colintkn — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -602,29 +602,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 | Priority | As a … | I want to … | So that I can…|
 | - | - | - | - |
-| `* * *` | new user | see usage instructions | refer to instructions if I forget how to use the app |
 | `* * *` | confused user | have a help command | learn how to use the application |
-| `* * *` | user | add a new person | |
-| `* * *` | user | add issues | |
-| `* * *` | user | see a list of pending tasks sorted by deadline | prioritise which to do |
-| `* * *` | user | search for tasks | |
+| `* * *` | user | add issues | track associated issues with rooms and their status|
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:606]: Might this be better phrased as

> track room issues and their status

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -602,29 +602,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* * *` | user | see a list of pending issues sorted by deadline | prioritise which issue to work on |
+| `* * *` | user | search for issues | find a specific issue and get its information |
+| `* * *` | user | search for issues | find issues matching a specific criteria |
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:609]: Hmm,

> search for issues

seems to be repeated.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -602,29 +602,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * *` | user | close issues | focus on open issues |
-| `* * *` | user | update issues | |
-| `* * *` | user | remove issues | |
-| `* * *` | user | add residents | |
-| `* * *` | user | search for residents | |
+| `* * *` | user | update issues | ensure it has the latest and correct information |
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:611]: How about

> ensure issues have the latest and correct information

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -602,29 +602,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| `* * *` | user | remove residents | stop tracking residents who are not in the hostel or don't wish to stay anymore |
+| `* * *` | user | add new rooms | track and manage rooms in the hostel |
 | `* * *` | user | see a list of rooms with their statuses | know which rooms are available |
-| `* * *` | user | search for rooms | |
-| `* * *` | user | update room details | |
-| `* * *` | user | remove rooms | |
+| `* * *` | user | search for rooms | find a specific room or rooms matching a specific criteria |
+| `* * *` | user | update room details | ensure a room's information is up to date |
+| `* * *` | user | remove rooms | stop tracking rooms which are not in the hostel |
 | `* *` | user | see both available rooms and unassigned residents side by side | easily match residents to rooms |
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:622]: Is this user story still relevant?

> see both available rooms and unassigned residents side by side

**Review by benedictkhoomw — COMMENTED** [review]
Looks quite good to me. I left some suggestions.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (DrWala) [reply docs/DeveloperGuide.md:609]: This was intentional. I wanted to show that you can either find 1 issue (for instance for room 10-111) or all rooms with cockroaches (matching a specific criteria). What do you think?

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (DrWala) [reply docs/DeveloperGuide.md:622]: It isnt as we cannot filter by unoccupied rooms. I will remove it.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (benedictkhoomw) [reply docs/DeveloperGuide.md:609]: Oh, then maybe distinguish them as
"search for issues by room number" and
"search for issues by description"

Comment (DrWala) [reply docs/DeveloperGuide.md]: Alright

**Pull request merged**